jujugui/static/gui/src/app/components/unit-details/unit-details.js

```diff
@@ -41,23 +41,55 @@ YUI.add('unit-details', function() {
           }}});
     },
 
+    /**
+      Build a HTML list from an array of ports and an IP address
+
+      @method _getAddressList
+      @returns {String} HTML of list
+    */
+    _getAddressList: function(address, ports) {
+      var list = '';
+      if (ports && ports.length > 0 && address) {
+        list = '<ul class="unit-details__list">';
+        for (var i in ports) {
+          list += `<li class="unit-details__list-item">`+
+            `<a href="http://${address}:${ports[i]}" target="_blank">`+
+              `${address}:${ports[i]}`+
+            `</a>`+
+          `</li>`;
+        }
+        list += '</ul>';
+      }
+      return list;
```

Comment:
With JSX you can have arbitrary HTML wherever you like, so instead of doing string concatenation you could just write the HTML. This will make the testing easier as well. Something like this:

```javascript
    _getAddressList: function(address, ports) {
      if (!ports && ports.length === 0 || !address) {
          return;
      }
      var items = [];
      for (var i in ports) {
        var href = `http://${address}:${ports[i]}`;
        items.push(<li class="unit-details__list-item">
          <a href={href} target="_blank">
            {address}:{ports[i]}
          </a>
        </li>);
      }
      return (
        <ul className="unit-details__list">
          {items}
        </ul>);
    },
},
```